libraries/AP_HAL_Linux/UARTDriver.cpp

```diff
@@ -570,7 +589,20 @@ int LinuxUARTDriver::_read_fd(uint8_t *buf, uint16_t n)
     ret = ::read(_rd_fd, buf, n);
     if (ret > 0) {
         BUF_ADVANCETAIL(_readbuf, ret);
+    } else {
+        switch (errno) {
+            case EAGAIN: 
+                /* Ignore EAGAIN that resulted from non-blocking read */
+                break;
+            case EPIPE:
```

Comment:
if we get EPIPE then shouldn't we close the fd? Won't we just loop trying to read otherwise?